src/gmp.c

```diff
@@ -23134,6 +23202,31 @@ gmp_xml_handle_end_element (/* unused */ GMarkupParseContext* context,
                   return;
                 }
             }
+          else if (create_target_data->krb5_credential_id
+                   && find_credential_with_permission
+                       (create_target_data->krb5_credential_id,
+                        &krb5_credential,
+                        "get_credentials"))
+            SEND_TO_CLIENT_OR_FAIL (XML_INTERNAL_ERROR ("create_target"));
+          else if (create_target_data->krb5_credential_id
+                   && krb5_credential == 0)
+            {
+              if (send_find_error_to_client
+                   ("create_target", "Credential",
+                    create_target_data->snmp_credential_id,
+                    gmp_parser))
+                {
+                  error_send_to_client (error);
+                  return;
+                }
+            }
+          else if (create_target_data->smb_credential_id
```

Comment:
Minor: This check could be before the SMB check on 23131, to skip the SMB and KRB processing when both are given.

Reply:
Fixed in [0e49e1e](https://github.com/greenbone/gvmd/pull/2332/commits/0e49e1ec1be168cb7aa179f28150dc0a93541d52)